corehq/apps/reports/generic.py

```diff
@@ -437,7 +437,7 @@ def _update_initial_context(self):
 
         def is_datespan(field):
             field_fn = to_function(field) if isinstance(field, basestring) else field
-            return issubclass(field_fn, DatespanFilter)
+            return issubclass(field_fn, DatespanFilter) and field_fn.is_viewable
```

Comment:
This report does have a datespan, but it shouldn't be editable.  The "save report" modal tries to let you set "last week", "last month", etc, but it should always be "last month".  Is this more comprehensible?  (undecided between "editable" and viewable")

```python
def is_editable_datespan(field):
    field_fn = to_function(field) if isinstance(field, basestring) else field
    return issubclass(field_fn, DatespanFilter) and field_fn.is_editable

has_editable_datespan = any([is_editable_datespan(field) for field in self.fields])
```

The whole business is a little messy - the standard reports strongly suggest that you have a normal, adjustable datespan.  This and the `LastMonthDatespanFilter` are both hacks to get around that.